MariuszWalis228879

rafiti12 opened the pull request:
MariuszWalis228879

tomaszsztajkowski committed:
rename class
MyAdapter -> DrawerDriverAdapter

tomaszsztajkowski committed:
DrawerDriverAdapter
set new start position after drawing line

rafiti12 committed:
Drawing in Main Window

rafiti12 committed:
figureScript2
Added a new pattern to draw

rafiti12 committed:
Added a new adapter
New adapter allows to choose style of the line, dotted presented in test.
+ minor fixes in SelectTestFigureOptionListener

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawerDriverAdapter.java:
```diff
@@ -4,14 +4,15 @@
  * driver adapter to drawer with several bugs.
  */
-public class MyAdapter extends DrawPanelController implements Job2dDriver {
+public class DrawerDriverAdapter extends DrawPanelController implements Job2dDriver {
```
Inheritance is not used (no super drawLine calls). Should be removed!

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawerDriverAdapter.java:
```diff
@@ -27,7 +28,8 @@ public void operateTo(int x, int y) {
 		line.setEndCoordinates(x, y);
 
-		drawLine(line);
+		DrawerFeature.getDrawerController().drawLine(line);
```
Direct coupling with DrawerFeature class is unnecessary.

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java:
```diff
@@ -0,0 +1,37 @@
+import edu.kis.powp.jobs2d.features.DrawerFeature;
+
+
+public class LineDrawerAdapter extends DrawPanelController implements Job2dDriver {
```
Similar issues as in the other adapter.

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java:
```diff
@@ -0,0 +1,37 @@
+
+public class LineDrawerAdapter extends DrawPanelController implements Job2dDriver {
+    private int startX = 0, startY = 0;
+    public ILine style;
```
should be private!

kowallus commented on src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java:
```diff
@@ -8,14 +8,25 @@
 	public void actionPerformed(ActionEvent e) {
-		FiguresJoe.figureScript1(driverManager.getCurrentDriver());
+		switch(figureToDraw){
+			case 1:
```
Constants or enums would be preferred.

kowallus reviewed (commented):
3 pts

Diagram: dependency between adapter and DrawPanelController is missing.

rafiti12 committed:
Command package
Added a new package with 1 interface and 2 empty classes

rafiti12 committed:
DriverCommand
Driver command now uses Job2dDriver instead inheriting from it

rafiti12 committed:
SetPositionCommand & OperateToCommand
SetPositionCommand and OperateToCommand can be used now

rafiti12 committed:
ComplexCommand
Addaed and implemented ComplexCommand class

rafiti12 committed:
ComplexCommand
Addaed and implemented ComplexCommand class

rafiti12 committed:
Factories
- added, implemented and tested new class that draws new factories
- added constructors to OperateToCommand and SetPositionCommand
- added new EnumShape class that implements new enum field 'Shape'
- SelectTestFigureOptionListener now uses Shape enum instead of int

kowallus commented on src/main/java/edu/kis/powp/command/ComplexCommand.java:
```diff
@@ -0,0 +1,20 @@
+import java.util.List;
+
+public class ComplexCommand implements DriverCommand{
+    List<DriverCommand> commandList;
```
Should be private!

kowallus commented on src/main/java/edu/kis/powp/command/Factories.java:
```diff
@@ -0,0 +1,26 @@
+import java.util.List;
+
+public class Factories {
+    public static List<DriverCommand> drawRectangle(int x, int y, int a, int b) {
```
Wrong name. The method doesn't draw.
It would be simpler if the factories would return DriverCommands

kowallus commented on src/main/java/edu/kis/powp/command/OperateToCommand.java:
```diff
@@ -0,0 +1,18 @@
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+public class OperateToCommand implements DriverCommand {
+    int x;
+    int y;
```
Should be private!

kowallus reviewed (commented):
4 pts
